PhoneGap/SwrvePlugin/platforms/ios/CDVAppDelegate+Swizzling.m

```diff
@@ -0,0 +1,52 @@
+#import <objc/runtime.h>
+#import "AppDelegate.h"
+#import "Swrve.h"
+#import "SwrvePlugin.h"
+
+static void swizzleMethod(Class class, SEL destinationSelector, SEL sourceSelector);
+
+@implementation CDVAppDelegate (Swrve)
+
++ (void)load {
+    swizzleMethod([CDVAppDelegate class],
+        @selector(application:didFinishLaunchingWithOptions:),
+        @selector(my_application:didFinishLaunchingWithOptions:));
+
+    swizzleMethod([CDVAppDelegate class],
+        @selector(application:didReceiveRemoteNotification:),
+        @selector(my_application:didReceiveRemoteNotification:));
+}
+
+- (BOOL)my_application: (UIApplication*)application
+            didFinishLaunchingWithOptions:(NSDictionary*)launchOptions {
+    NSNumber *appId = NSBundle.mainBundle.infoDictionary[@"AppId"];
+    NSString *apiKey = NSBundle.mainBundle.infoDictionary[@"ApiKey"];
+    SwrveConfig* config = [[SwrveConfig alloc] init];
+    [SwrvePlugin initWithAppID:[appId intValue]
+                        apiKey:apiKey
+                        config:config
+                viewController:self.viewController
+                launchOptions:launchOptions];
+
+     return [self my_application:application didFinishLaunchingWithOptions:launchOptions];
+}
+
+-(void) my_application: (UIApplication*) application didReceiveRemoteNotification:(NSDictionary *)userInfo {
```

Comment:
This method enters into an infinite loop and crashes the app. It looks like no class in the hierarchy implements this method so it can't be swizzled.

A potential solution would be to save a flag to detect if this method has to call any implementation that was there before and check before the last statement.